### ISSUE TITLE
Dose NCAPK-package work with/accept the data set that includes both day and nominal time?

### ISSUE THREAD

**Issue by cokkies** [billdenney/pknca#49]
Hallo every body

I'm quite new in this scientific forum. I just would like to know that anyone has a answer to the titled question? How do I organize the R-code by a way, so I can calculate the NCAPK-parameters regarding subject, or group, or day, or dose? I'm appreciated for your constructive inputs.... 

![sample picture](https://user-images.githubusercontent.com/32448246/31068342-e093f1de-a757-11e7-9b01-236553672a17.PNG)

**Comment by billdenney** [comment]
Hi @cokkies,

The direct answer to your question is that you need to make a single column to define the time.  For that, you'd need to do something like:

mydata$Time_Nominal <- (Study.Day - 1)*24 + Hour.Nominal

Generally, PKNCA also does not impute time zero so it looks like you would also need to do that before running PKNCA.

**Comment by cokkies** [comment]
Hi @billdenney 

Do you suggest that the time points = 0 and the corresponding concentration data should be removed from the data set before PKNCA run, even there exist non-zero concentration records when time = 0?

**Comment by billdenney** [comment]
Sorry for the confusion: I was suggesting the opposite.  If you don't have a time=0 measurement, you will probably need to add it.

For instance, subject R0125 does not appear to have a Day 1, hour 0 measurement in the data you inserted above.  If you want an AUC0-24, you will need to include subject R0125, day 1, hour 0, concentration= (something, likely 0) in your dataset.

**Comment by cokkies** [comment]
Thanks, @billdenney. As you suggested, I did add time =0 & ID as concentration =0 in the data set. My data set contains a serveral subjects testing against different doses in different days. 
Then I wrote some syntax in R-script like that:
library(knitr)
library(PKNCA)
conc.data<- read.csv("C:/PK_excel/dataset.csv", stringsAsFactors=FALSE, header=TRUE)
knitr::kable(head(conc.data))
my.conc <- PKNCAconc(as.data.frame(conc.data), DV~TIME|ID)
then I did experience a issue like that: "Error in PKNCAconc.data.frame(as.data.frame(conc.data), DV ~ TIME | ID) : 
Rows that are not unique per group and time (column names: TIME, ID) found within concentration data."
d.dose <- unique(conc.data[conc.data$TIME == 0,c("DOSE", "TIME", "ID")]) 
knitr::kable(d.dose,caption="Example dosing data extracted from concerned data set")

![sample picture_1](https://user-images.githubusercontent.com/32448246/31376132-347e0a26-ada4-11e7-9774-bee6ea1bc0b7.PNG)

my.dose <- PKNCAdose(d.dose, DOSE~TIME|ID) 
then I did experience a issue like that: "Error in PKNCAdose.data.frame(d.dose, DOSE ~ TIME | ID) : 
  Rows that are not unique per group and time (column names: TIME, ID) found within dosing data."

Any solution for the issue?
